### PR TITLE
Hb 06

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: src/main/java/com/dev/cinema/model/Ticket.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@Table(name = "ticket")
+public class Ticket {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private MovieSession movieSession;
+    @OneToOne
+    private User user;
+    @ManyToOne()
```
Comment (ilya449) [inline src/main/java/com/dev/cinema/model/Ticket.java]: extra "()"

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    ShoppingCartDao shoppingCartDao;
+    @Inject
+    TicketDao ticketDao;
```
Comment (ilya449) [inline src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: access modifiers?

Comment (vremere-a) [reply src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/com/dev/cinema/model/Ticket.java (hunk 1 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/Ticket.java]: done

File: src/main/java/com/dev/cinema/model/Ticket.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+public class Ticket {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private MovieSession movieSession;
+    @OneToOne
+    private User user;
+    @ManyToOne
+    private ShoppingCart shoppingCart;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/Ticket.java]: we don't need this relation here, ticket shouldn't know about shopping cart

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public void registerNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartDao.add(shoppingCart);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.getTickets().removeAll(shoppingCart.getTickets());
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: why not list clear?

File: src/main/java/com/dev/cinema/model/Ticket.java (hunk 3 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/Ticket.java]: ok

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: ok

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**